### PR TITLE
js for checking boxes and checkbox positioning

### PR DESCRIPTION
**Pull request by straleyb**
Screen Shot 2019-08-20 at 1 44 27 PM
Screen Shot 2019-08-20 at 1 44 11 PM

### CODE REVIEW

File: app/assets/javascripts/oregon_digital/add_to_my_shelf.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+$( document ).ready(function() {
+  $('#Select_all_on_page').click(function() {
+    $('.my_shelf_result_checkbox').each(function () { this.checked = !this.checked; });
```
Comment (CGillen) [inline app/assets/javascripts/oregon_digital/add_to_my_shelf.js]: This doesn't seem to work as expected for a mix of already checked and unchecked boxes

File: app/views/_controls.html.erb [hunk 2]
```diff
@@ -5,9 +5,9 @@
         <%= render partial: 'catalog/search_form' %>
       </div>
       <div class="col-md-offset-10">
-        <%= button_to "Add results to My Shelf", root_path, class: 'btn btn-primary my-shelf-button' %>
-        <%= label_tag "Add All on page to My Shelf" do %>
-          <% raw("#{check_box_tag 'Add All on page to My Shelf'} Add All on page to My Shelf") %>
+        <%= button_to "Add to My Shelf", root_path, class: 'btn btn-primary my-shelf-button' %>
+        <%= label_tag "Add to My Shelf" do %>
```
Comment (CGillen) [inline app/views/_controls.html.erb]: The label is for the checkbox, not the button. `<%= label_tag "" do %>`

File: app/assets/stylesheets/oregon_digital/_search.scss [hunk 3]
```diff
@@ -4,4 +4,9 @@
 
 .my-shelf-button {
   margin-top: 15px;
+}
+
+.shift-right {
```
Comment (CGillen) [inline app/assets/stylesheets/oregon_digital/_search.scss]: This is problematic with long titles
image

Maybe we can work it into the resource-type div and simplify it to a `class="pull-right"`?

Comment (straleyb) [reply app/assets/stylesheets/oregon_digital/_search.scss]: I tried pull right but it didnt seem to operate as expected.

File: app/assets/javascripts/oregon_digital/add_to_my_shelf.js (hunk 1 above)
Comment (straleyb) [reply app/assets/javascripts/oregon_digital/add_to_my_shelf.js]: Yeah so if some are checked, and some arent, we check all the unchecked ones and leave the others. If all are checked, then uncheck them. That sound about right?

Comment (CGillen) [reply app/assets/javascripts/oregon_digital/add_to_my_shelf.js]: Sounds right, but complicated. I think we get the same outcome by setting all .my_shelf_result_checkbox checked values to #Select_all_on_page checked value

Comment (straleyb) [reply app/assets/javascripts/oregon_digital/add_to_my_shelf.js]: Yeah I see what you are saying. we should be able to push the #Select_all_on_page object through then check its state. 👍

File: app/assets/stylesheets/oregon_digital/_search.scss (hunk 3 above)
Comment (CGillen) [reply app/assets/stylesheets/oregon_digital/_search.scss]: pushed 7b547ab as a fix for works with long names

image

**Pull request merged**